### DISCUSSION TITLE
Could you tell me how to use aug_seq at mx.io.ImageRecordIter?

### DISCUSSION THREAD

**Discussion by ghost** [apache/incubator-mxnet#12671] [Q&A]
## Description
I was wondering if there are same examples for us to know how to use aug_seq at mx.io.ImageRecordIter. there are just a few information for it.

## Environment info (Required)
```
linux
GPU
```
Package used (Python/R/Scala/Julia):
I'm using Python for tackle the problem.
Python 2.7

**Comment by vandanavk** [comment]
@mxnet-label-bot [Question, Data Loading]

**Comment by thomelane** [comment]
Hi @shawncheeer,

As I understand things, `aug_seq` should be a string referring to the augmenter you want to use.

An 'augmenter' here, isn't actually a single type of augmentation (e.g. resize) but instead it's a `ImageAugmenter` class with a `Process` method that performs all the augmentation steps required (e.g. resize, colour shifts, crops). 

MXNet has a two augmenter out of the box called `aug_default` and `det_aug_default ` which are the defaults for `ImageRecordIter` and `ImageDetRecordIter` respectively. Check out https://github.com/apache/incubator-mxnet/blob/master/src/io/image_aug_default.cc to see how `aug_default` is implemented. And you can use this as a starting point for your own custom augmenters.

One last thing to note is that before you can reference your own custom augmenter in `aug_seq`, it needs to registered. See https://github.com/apache/incubator-mxnet/blob/master/src/io/image_aug_default.cc#L571 for an example of how `aug_default` is registered: allowing it to be referenced in `aug_seq`.

@tqchen could you confirm that the above is correct? thanks!

Sources: https://github.com/apache/incubator-mxnet/pull/1956

**Comment by thomelane** [comment]
Aside from the above, I'd take a look at Gluon's Transforms. You'll probably find they're an easier way to implement what you need. See https://mxnet.incubator.apache.org/api/python/gluon/data.html#vision-transforms
